analytics/analytics.go

```diff
@@ -348,3 +349,23 @@ func (a *AnalyticsRecord) IsGraphRecord() bool {
 
 	return false
 }
+
+func (a *AnalyticsRecord) RemoveIgnoredFields(ignoreFields []string) {
```

Comment:
@mativm02 I think using reflection here makes this change way more dangerous. Can you try to refactor it using `github.com/fatih/structs`? We already use that lib in this type of stuff (for example https://github.com/TykTechnologies/tyk-pump/blob/1f0bc24934cd6d09829f5c0f6d6fe2e243f6b7ac/analytics/aggregate.go#L160), and IMO it's much safer

Reply:
Nope @mativm02 - we didn't have any problem related to that repo (yet - I don't want to jinx it 😅 )

Reply:
@tbuchaillot done! now it's implementing `structs` library